go/viperutil/internal/sync/sync_test.go

```diff
@@ -75,19 +72,22 @@ func TestWatchConfig(t *testing.T) {
 
 		return nil
 	}
-	writeRandomConfig := func() error {
+	writeRandomConfig := func(tmp *os.File) error {
 		a, b := rand.Intn(100), rand.Intn(100)
-		return writeConfig(a, b)
+		return writeConfig(tmp, a, b)
 	}
 
-	require.NoError(t, writeRandomConfig())
+	tmp, err := os.CreateTemp(t.TempDir(), "TestWatchConfig_*.json")
```

Comment:
Noting that this is the key fix, as we're using a dedicated directory for our config files rather than the (very deep) PWD for the tests of the vitess root dir.